File: 2024/01/01.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func abs(x int) int {
	if x > 0 {
		return x
	}
	return -x
}

func main() {
	f, _ := os.Open("input")
	scanner := bufio.NewScanner(f)
	defer f.Close()

	var left, right []int

	idCount := make(map[int]int)

	for scanner.Scan() {
		var l_id, r_id int
		fmt.Sscanf(scanner.Text(), "%d %d", &l_id, &r_id)
		left = append(left, l_id)
		right = append(right, r_id)
		idCount[r_id]++
	}

	sort.Ints(left)
	sort.Ints(right)

	diff := 0
	for i := range left {
		diff += abs(left[i] - right[i])
	}

	fmt.Println(diff)

	similarity := 0
	for _, value := range left {
		similarity += idCount[value] * value
	}

	fmt.Println(similarity)
}

File: 2024/01/01.py
lst_one = []
lst_two = []

for _ in range(1000):
    in_one, in_two = map(int, input().split())
    lst_one.append(in_one)
    lst_two.append(in_two)

lst_one.sort()
lst_two.sort()

counter = 0

for idx, item in enumerate(lst_one):
    counter += abs(item - lst_two[idx])

print(counter)

from collections import Counter

similarity = 0

items_counter = Counter(lst_two)

for idx, item in enumerate(lst_one):
    similarity += item * items_counter[item]

print(similarity)
